pkg/stacks/relationship.go

```diff
@@ -48,16 +61,54 @@ type KindlyIdentifier interface {
 	GroupVersionKind() schema.GroupVersionKind
 }
 
+// MultiParentLabelPrefix returns the NS specific prefix of a multi-parent label
+// for resources co-owned by a set of Stacks.
+//
+// This prefix is suitable for identifying resources labeled within a given
+// namespace. The prefix may include a predictable truncation suffix if the
+// namespace exceeds 32 characters. This truncation length permits another 32
+// characters for a (potentially truncated) resource name to be appended to the
+// label.
+func MultiParentLabelPrefix(stackParent metav1.Object) string {
+	ns := stackParent.GetNamespace()
+
+	// guaranteed not to error based on the lengths of values we are supplying
```

Comment:
What if a future maintainer changes the lengths?

Reply:
I've left a warning on preserveNSLength :+1: